modules/Bio/EnsEMBL/Compara/DBSQL/AlignSliceAdaptor.pm

```diff
@@ -433,111 +381,101 @@ sub _combine_genomic_align_trees {
 
     my $match = 0;
     ## SECONDARY LOOP. Note that the $species_counter is not reset at the end of the loop.
-    ## This ensures that we do not add two nodes to the same species_order track and that we
-    ## preserve the order in all existing and in the new tree.
-    while (!$match and $species_counter < @$species_order) {
-      my $species_genome_db = $species_order->[$species_counter]->{genome_db};
-      my $species_right_node_id = $species_order->[$species_counter]->{right_node_id};
+    ## This ensures that we do not add two nodes to $species_order and that we preserve as
+    ## far as possible the order in all existing and in the new trees.
+    while (!$match) {
+      my $species = $species_order->[$species_counter];
+      my ($species_genome_db, $species_right_node_id);
+      $species_genome_db = $species->{genome_db} if defined $species;
+      $species_right_node_id = $species->{right_node_id} if defined $species;
       $match = 1;
 
-      ## 1. Use info from species_right_node_id if available
+      ## 1. Use info from $species_right_node_id if available
       if (defined($species_right_node_id) and $species_right_node_id == $this_node_id) {
-          $species_order->[$species_counter]->{right_node_id} = $this_right_node_id;
-          # #         $species_order->[$species_counter]->{last_node} = $this_genomic_align_node;
+        $species_order->[$species_counter]->{right_node_id} = $this_right_node_id;
```

Comment:
There are a few of these git blame/white space change issues floating about. I don't mind personally, but I think we are supposed to preserve where possible.

Reply:
I know, and I tried, but they really did my work harder to track and follow (2 spaces as indentation is already hard enough :confounded:), so I'm willing to take the blame :nerd_face: